test/jdk/javax/swing/JMenu/4213634/bug4213634.java

```diff
@@ -39,85 +39,77 @@
  * @test
  * @key headful
  * @bug 4213634 8017187
- * @author Scott Violet
  * @library ../../regtesthelpers
  * @build Util
  * @run main bug4213634
  */
 
-
 public class bug4213634 {
 
-    private JMenu menu;
-
-    private JFrame frame;
-
-    public static void main(String[] args) throws Throwable {
-        new bug4213634();
-    }
+    private static JMenu menu;
+    private static JFrame frame;
+    private static Robot robot;
 
-    bug4213634() throws AWTException, InterruptedException, InvocationTargetException {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
-                createAndShowGUI();
-            }
-        });
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            SwingUtilities.invokeAndWait(() -> createAndShowGUI());
 
-        test();
+            robot.waitForIdle();
+            robot.delay(1000);
+            test();
+        } finally {
+            SwingUtilities.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
     }
 
-    public  void createAndShowGUI() {
-        frame = new JFrame("TEST");
+    public static void createAndShowGUI() {
+        frame = new JFrame("bug4213634");
         JMenuBar mb = new JMenuBar();
         menu = mb.add(createMenu("1 - First Menu", true));
         mb.add(createMenu("2 - Second Menu", false));
         frame.setJMenuBar(mb);
-        JTextArea ta = new JTextArea("This test dedicated to Nancy and Kathleen, testers and bowlers extraordinaire\n\n\nNo exception means pass.");
-        frame.getContentPane().add("Center", ta);
         JButton button = new JButton("Test");
         frame.getContentPane().add("South", button);
-        frame.setBounds(100, 100, 400, 400);
+        frame.setSize(300, 300);
+        frame.setLocationRelativeTo(null);
         frame.setVisible(true);
         button.requestFocusInWindow();
```

Comment:
Since the root cause is unknown and you are trying to make the test more robust, should we add a delay after `frame.setVisible(true);` ? I'm just suspecting request focus failure after `setVisible`.

Reply:
that's why I added waitForIdle/delay after createShowGUI

Reply:
I meant in right after `frame.setVisible(true);`, between line 80 and 81.

Reply:
I know what you meant but I am saying its not needed as historically we added delay/waitForIdle after we do setVisible and start the test..
Also requestFocusInWindow is not guranteed to be done by JDK, its just a request...

Reply:
Now that the text component is gone, the frame has the only focusable component which is the button. So the button should get the focus even without the request.